src/DynamoApplications/StartupUtils.cs

```diff
@@ -0,0 +1,302 @@
+            .Add("s=|S=", "PresetStateID, Instruct Dynamo to set the graph to the specified preset by name,"+
+            "this can be set to a statename or 'all', which will evaluate all states in the dyn", s => presetStateid = s)
+            .Add("v=|V=", "Verbose, Instruct Dynamo to output all evalautions it performs to an xml file at this path", v => verbose = v);
+
+            optionsSet.Parse(args);
+
+            //check for incompatabile parameters
+            if ((!string.IsNullOrEmpty(presetStateid) || (!string.IsNullOrEmpty(presetFile) || (!string.IsNullOrEmpty(verbose)))) && string.IsNullOrEmpty(openfilepath))
+            {
+                Console.WriteLine("you must supply a file to open if you want to load a preset or presetFile, or want to save an evaluation output ");
+                
+            }
+
+
+            return new CommandLineArguments
+            {
+                Locale = locale,
+                CommandFilePath = commandFilePath,
+                OpenFilePath = openfilepath,
+                PresetStateID = presetStateid,
+                PresetFilePath = presetFile,
+                Verbose = verbose,
+            };
+        }
+
+        public string Locale { get; set; }
+        public string CommandFilePath { get; set; }
+        public string OpenFilePath { get; set; }
+        public string PresetStateID { get; set; }
+        public string PresetFilePath { get; set; }
+        public string Verbose { get; set; }
+    }
+
+    public class Preloading
+    {
+        public static void PreloadShapeManager(ref string geometryFactoryPath, ref string preloaderLocation)
+        {
+            var exePath = Assembly.GetExecutingAssembly().Location;
+            var rootFolder = Path.GetDirectoryName(exePath);
+
+            var versions = new[]
+            {
+                LibraryVersion.Version219,
+                LibraryVersion.Version220,
+                LibraryVersion.Version221
+            };
+
+            var preloader = new Preloader(rootFolder, versions);
+            preloader.Preload();
+            geometryFactoryPath = preloader.GeometryFactoryPath;
+            preloaderLocation = preloader.PreloaderLocation;
+        }
+
+        public static DynamoModel MakeModel(bool CLI, IEnumerable<string> librariesToPreload)
+        {
+
+            var geometryFactoryPath = string.Empty;
+            var preloaderLocation = string.Empty;
+            PreloadShapeManager(ref geometryFactoryPath, ref preloaderLocation);
+
+            var config = new DynamoModel.DefaultStartConfiguration()
+                  {
+                      PathResolver = new PathResolver(preloaderLocation, librariesToPreload),
+                      GeometryFactoryPath = geometryFactoryPath,
+                      StartInTestMode = true
+                  };
+
+            //if we are building a model for CLI mode, then we don't want to start an updateManager
```

Comment:
Move these comments into doc comments on the `InitializeUpdateManager` method proposed below.